### PR TITLE
Fix git workflow documentation

### PR DESCRIPTION
**Pull request by SammyVimes**
### Changelog category 

Documentation (changelog entry is not required)

### CODE REVIEW

File: ydb/docs/en/core/development/suggest-change.md [hunk 1]
```diff
@@ -98,43 +98,40 @@ And copy-paste the shown token to complete the GitHub CLI configuration.
 
 YDB official repository is [https://github.com/ydb-platform/ydb](https://github.com/ydb-platform/ydb), located under the YDB organization account `ydb-platform`.
 
-Create working dir:
+To work on the YDB code changes, you need to create a fork repository under your GitHub account. Create a fork pressing `Fork` button on the official repository page.
```
Comment (blinkov) [inline ydb/docs/en/core/development/suggest-change.md:101]:
```suggestion
To work on the {{ ydb-short-name }} code changes, you need to create a fork repository under your GitHub account. Create a fork by pressing the `Fork` button on the [official {{ ydb-short-name }} repository page](https://github.com/ydb-platform/ydb).
```

File: ydb/docs/en/core/development/suggest-change.md [hunk 2]
```diff
@@ -98,43 +98,40 @@ And copy-paste the shown token to complete the GitHub CLI configuration.
...
+
+After your fork is set up, create a local git repository with two remotes:
+- official: official YDB repository, for main and stable branches
```
Comment (blinkov) [inline ydb/docs/en/core/development/suggest-change.md:104]: why not something like upstream/origin? these names are more common

File: ydb/docs/en/core/development/suggest-change.md [hunk 3]
```diff
@@ -98,43 +98,40 @@ And copy-paste the shown token to complete the GitHub CLI configuration.
...
+- fork: your YDB repository fork, for your development branches
```
Comment (blinkov) [inline ydb/docs/en/core/development/suggest-change.md:105]:
```suggestion
- `official`: official {{ ydb-short-name }} repository, for main and stable branches
- `fork`: your {{ ydb-short-name }} repository fork, for your development branches
```

File: ydb/docs/en/core/development/suggest-change.md [hunk 4]
```diff
@@ -98,43 +98,40 @@ And copy-paste the shown token to complete the GitHub CLI configuration.
...
-There's a single GitHub CLI command which does all of that together:
-
 ```
-gh repo fork ydb-platform/ydb --default-branch-only --clone
+cd ydb
+git remote add fork git@github.com:{your_github_user_name}/ydb.git
 ```
 
-Once completed, you have a YDB Git repository fork cloned to `~/ydbwork/ydb`.
+Once completed, you have a YDB Git repository set up in `~/ydbwork/ydb`.
```
Comment (blinkov) [inline ydb/docs/en/core/development/suggest-change.md:118]:
```suggestion
Once completed, you have a {{ ydb-short-name }} Git repository set up in `~/ydbwork/ydb`.
```

File: ydb/docs/en/core/development/suggest-change.md [hunk 5]
```diff
@@ -98,43 +98,40 @@ And copy-paste the shown token to complete the GitHub CLI configuration.
...
+
+Forking a repository is an instant action, however cloning to the local machine takes some time to transfer about 650 MB of repository data over the network.
 
-- git
+Next setup default git push behaviour:
```
Comment (blinkov) [inline ydb/docs/en/core/development/suggest-change.md:122]:
```suggestion
Next, let's configure the default `git push` behaviour:
```

File: ydb/docs/en/core/development/suggest-change.md [hunk 6]
```diff
@@ -98,43 +98,40 @@ And copy-paste the shown token to complete the GitHub CLI configuration.
...
 ```
-{% endlist %}
 
-Forking a repository is an instant action, however cloning to the local machine takes some time to transfer about 650 MB of repository data over the network.
+This way, `git push {remote}` command will automatically set upstream for the current branch to the `{remote}` and consecutive `git push` commands will only push current branch.  
 
 ### Configure commit authorship {#author}
 
-Run the following command from your repository directory to set up your name and email for commits pushed using Git:
+Run the following command to set up your name and email for commits pushed using Git:
```
Comment (blinkov) [inline ydb/docs/en/core/development/suggest-change.md:133]:
```suggestion
Run the following command to set up your name and email for commits pushed using Git (replace example name and email with your real ones):
```

File: ydb/docs/en/core/development/suggest-change.md [hunk 7]
```diff
@@ -146,29 +143,29 @@ To start working on a feature, ensure the steps specified in the [Setup the envi
 
 ### Refresh trunk {#fork_sync}
 
-Usually you need a fresh trunk revision to branch from. Sync fork to obtain it, running the following command:
-
-```
-gh repo sync your_github_login/ydb -s ydb-platform/ydb
-```
+Usually you need a fresh trunk revision to branch from. Sync your local main branch running the following command in the repository:
```
Comment (blinkov) [inline ydb/docs/en/core/development/suggest-change.md:146]:
```suggestion
Usually you need a fresh revision to branch from. Sync your local `main` branch by running the following command in the repository:
```

File: ydb/docs/en/core/development/suggest-change.md [hunk 8]
```diff
@@ -146,29 +143,29 @@ To start working on a feature, ensure the steps specified in the [Setup the envi
...
 
-This statement performs sync remotely on GitHub. Pull the changes to the local repository then:
+If your current local branch is not `main`:
 
 ```
 cd ~/ydbwork/ydb
+git fetch official main:main
 ```
+
+This updates your local main branch without checking it out.
```
Comment (blinkov) [inline ydb/docs/en/core/development/suggest-change.md:155]:
```suggestion
This command updates your local `main` branch without checking it out.
```

File: ydb/docs/en/core/development/suggest-change.md [hunk 9]
```diff
@@ -146,29 +143,29 @@ To start working on a feature, ensure the steps specified in the [Setup the envi
...
+
+If your current local branch is `main`:
```
Comment (blinkov) [inline ydb/docs/en/core/development/suggest-change.md:157]: I'd swap this example with the previous one

File: ydb/docs/en/core/development/suggest-change.md [hunk 10]
```diff
@@ -146,29 +143,29 @@ To start working on a feature, ensure the steps specified in the [Setup the envi
...
 ```
-git checkout main
-git pull
+git pull --ff-only official main
 ```
 
 ### Create a development branch {#create_devbranch}
 
-Create a development branch using Git (replace "feature42" with your branch name), and assign upstream for it:
+Create a development branch using Git (replace "feature42" with your branch name):
```
Comment (blinkov) [inline ydb/docs/en/core/development/suggest-change.md:165]:
```suggestion
Create a development branch using Git (replace "feature42" with a name for your new branch):
```

File: ydb/docs/en/core/development/suggest-change.md [hunk 11]
```diff
@@ -177,32 +174,52 @@ Edit files locally, use standard Git commands to add files, verify status, make
...
+  remote: Create a pull request for '{branch_name}' on GitHub by visiting:
+  remote:      https://github.com/{your_github_user_name}/test/pull/new/{branch_name}
+  ... 
+  ```
+
+- GitHub CLI
+
+  ```
+  cd ~/ydbwork/ydb
+  ```
```
Comment (blinkov) [inline ydb/docs/en/core/development/suggest-change.md:214]:
```suggestion
  Install and configure [GitHub CLI](https://cli.github.com/).
  ```
  cd ~/ydbwork/ydb
  ```
```

File: ydb/docs/en/core/development/suggest-change.md [hunk 12]
```diff
@@ -232,11 +249,28 @@ If there's a Pull Request opened for some development branch in your repository,
...
+```
+
+Then cherry-pick the fix and push the branch to your fork:
+
+```
+git cherry-pick {fixes_commit_hash}
+git push fork
+```
+
+And then create a PR from your branch with the cherry-picked fix to the stable branch.
```
Comment (blinkov) [inline ydb/docs/en/core/development/suggest-change.md:276]:
```suggestion
And then create a PR from your branch with the cherry-picked fix to the stable branch. It is done similarly to opening a PR to `main`, but make sure to double-check the target branch.
```

**Review by blinkov — CHANGES_REQUESTED** [review]

**Review by alexv-smirnov — APPROVED** [review]

File: ydb/docs/en/core/development/suggest-change.md (hunk 2 above)
Comment (SammyVimes) [reply ydb/docs/en/core/development/suggest-change.md]: This way we have non-default remote names for non-default scenario (default scenario is just one remote -- origin). Thus developers won't be able to (unconsciously) push somewhere they did not intend to

**Review by fomichev3000 — APPROVED** [review]

**Review by blinkov — APPROVED** [review]

**Pull request merged**